packages/eas-cli/src/commands/channel/delete.ts

```diff
@@ -0,0 +1,164 @@
+import { getConfig } from '@expo/config';
+import { flags } from '@oclif/command';
+import { error } from '@oclif/errors';
+import gql from 'graphql-tag';
+
+import EasCommand from '../../commandUtils/EasCommand';
+import { graphqlClient, withErrorHandlingAsync } from '../../graphql/client';
+import {
+  DeleteUpdateChannelMutation,
+  DeleteUpdateChannelMutationVariables,
+  DeleteUpdateChannelResult,
+  GetChannelInfoQuery,
+  GetChannelInfoQueryVariables,
+} from '../../graphql/generated';
+import Log from '../../log';
+import {
+  findProjectRootAsync,
+  getProjectFullNameAsync,
+  getProjectIdAsync,
+} from '../../project/projectUtils';
+import { promptAsync, toggleConfirmAsync } from '../../prompts';
+import { enableJsonOutput, printJsonOnlyOutput } from '../../utils/json';
+
+export default class ChannelDelete extends EasCommand {
+  static hidden = true;
+  static description = 'Delete a channel on the current project';
+
+  static args = [
+    {
+      name: 'name',
+      required: false,
+      description: 'Name of the channel to delete',
+    },
+  ];
+  static flags = {
+    json: flags.boolean({
+      description: `Delete a channel on the current project`,
+      default: false,
+    }),
+    'non-interactive': flags.boolean({
+      default: false,
+      description: 'Run command in non-interactive mode',
+    }),
+  };
+
+  async runAsync(): Promise<void> {
+    const {
+      args: { name: nameArg },
+      flags: { json: jsonFlag, 'non-interactive': nonInteractiveFlag },
+    } = this.parse(ChannelDelete);
+    if (jsonFlag && !nonInteractiveFlag) {
+      error('--json is allowed only in non-interactive mode', { exit: 1 });
+    }
```

Comment:
There is an argument that setting `--json` should automatically set `nonInteractiveFlag = true`, but for now am following the convention set here: https://github.com/expo/eas-cli/blob/main/packages/eas-cli/src/commands/build/index.ts#L107